app/src/pages/[resourceType]/[id].tsx

```diff
@@ -27,14 +33,49 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     return encodedCql ? Buffer.from(encodedCql, 'base64').toString() : null;
   }, [jsonData]);
 
+  const ctx = trpc.useContext();
+
+  const draftMutation = trpc.draft.createDraft.useMutation({
+    onSuccess: data => {
+      notifications.show({
+        title: `${jsonData.resourceType} Cloned!`,
+        message: `${jsonData.resourceType} successfully cloned`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      router.push(`/authoring/${jsonData.resourceType}/${data.draftId}`);
+      ctx.draft.getDraftCounts.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: `${jsonData.resourceType} Clone Failed!`,
+        message: `Attempt to clone ${jsonData.resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  const router = useRouter();
+
+  const cloneResource = () => {
+    const newClonedResource = jsonData;
+    newClonedResource.id = uuidv4();
+    newClonedResource.status = 'draft';
+    draftMutation.mutate({ resourceType, draft: newClonedResource });
+  };
+
   return (
     <div>
       <Stack spacing="xs">
         <div>
-          <Group>
+          <Group position="apart">
             <Text size="xl" color="gray">
               {jsonData.resourceType}/{jsonData.id}
             </Text>
+            <Button w={240} loading={draftMutation.isLoading} onClick={cloneResource}>
+              Clone to Draft Artifact
```

Comment:
I think the nomenclature might get confusing here. [Clone](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#clone) is a separate operation than draft, which is very similar. Maybe instead of the word `clone` we can say something else.

* Create Draft of <some identifying string>
* Create Draft of <resource type>
* Something else?

Reply:
Hmmm...I think I'll go with Create Draft of 😆

Reply:
lol GitHub escaped my comment. I edited it

Reply:
For the "Cloned!" and "Successfully cloned!" messages, can I just change them to "Created!" and "Successfully created" or should I distinguish them from the create draft from scratch somehow?